test/e2e/containerd/freeze_test.go

```diff
@@ -0,0 +1,225 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2022 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package containerd
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/environment"
+	"knative.dev/serving/test"
+)
+
+const (
+	ksvcUrl         = "sleeptalker.default.example.com"
+	deployNamespace = "default"
+	workerNodeName  = "kind-worker"
+	lbsvcNamespace  = "kourier-system"
+	lbsvcName       = "kourier"
+)
+
+func newClient() (*test.Clients, error) {
+	env := environment.ClientConfig{}
+	flag.Parse()
+	cfg, err := env.GetRESTConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get kubeconfig %v", err)
+	}
+
+	clients, err := test.NewClients(cfg, "default")
+	if err != nil {
+		return nil, fmt.Errorf("failed to setup clients: %v", err)
+	}
+
+	return clients, nil
+}
+
+func findTheLastTimestamp(logs []byte) (time.Time, error) {
+	lines := strings.Split(string(logs), "\n")
+	//The last line has \n, so last array element is empty, we choose the penultimate element
+	lastLine := lines[len(lines)-2]
```

Comment:
I'm a little leery of hardcoding this in case the behavior changes in the future... what if we looked through the array (starting at the last element) and stopped on the first one that was had an actual log?

Reply:
fixed, just take the last 3 lines:
```go
var logTailNums int64 = 3
	sleepTalkerPod := pods.Items[0]
	tickLogsByte, err := client.CoreV1().Pods(deployNamespace).
		GetLogs(sleepTalkerPod.Name, &v1.PodLogOptions{Container: "user-container", TailLines: &logTailNums}).
		Do(ctx).Raw()
```